third_party/intel/lib/TritonIntelGPUToLLVM/TritonOpsToLLVM.cpp

```diff
@@ -282,8 +282,17 @@ class LoadStorePrefetchOpConversion
     if constexpr (std::is_same_v<OpType, LoadOp>) {
       rewriter.restoreInsertionPoint(insertPoint);
 
-      TritonGEN::SIMDBlockReadOp simdRead =
-          rewriter.create<TritonGEN::SIMDBlockReadOp>(loc, v64i16Ty, base);
+      VectorType v8i16Ty = VectorType::get(8, i16_ty);
+      SmallVector<Value> values;
+      Value offset = i32_val(128);
+      for (int i = 0; i < 8; ++i) {
+        auto simdRead =
+            rewriter.create<TritonGEN::SIMDBlockReadOp>(loc, v8i16Ty, base);
+        values.push_back(simdRead.getRes());
+        base = gep(ptrToSharedMemTy, i16_ty, base, offset);
+      }
+      auto simdRead = rewriter.create<GlueOp>(loc, v64i16Ty, values);
```

Comment:
```suggestion
      VectorType v8i16Ty = VectorType::get(8, i16_ty);
      SmallVector<Value> values;
      Value offset = i32_val(128);
      for (int i = 0; i < 8; ++i) {
        auto simdRead =
            rewriter.create<TritonGEN::SIMDBlockReadOp>(loc, v8i16Ty, base);
        values.push_back(simdRead.getRes());
        base = gep(ptrToSharedMemTy, i16_ty, base, offset);
      }
      auto simdRead = rewriter.create<GlueOp>(loc, v64i16Ty, values);
```
```suggestion
      VectorType v8i16Ty = VectorType::get(8, i16_ty);
      SmallVector<Value> values;
      Value offset = i32_val(/*getSubGroupSize()*/);
      for (int i = 0; i < 8; ++i) {
        auto simdRead =
            rewriter.create<TritonGEN::SIMDBlockReadOp>(loc, v8i16Ty, base);
        values.push_back(simdRead.getRes());
        base = gep(ptrToSharedMemTy, v8i16Ty, base, offset);
      }
      auto simdRead = rewriter.create<GlueOp>(loc, v64i16Ty, values);
```

Equivalent and more intuitive. This is a sub-group operation in which each work-item reads a `v8i16Ty` value, so we can offset by sub-group size `v8i16Ty` elements.

Reply:
Also, can we avoid magic number `8`?

Reply:
Ups, that was weird.

Reply:
done.